src/index.rs

```diff
@@ -515,17 +514,14 @@ fn discover_paths<P: AsRef<Path>>(
         .collect()
 }
 
-fn scan_entry(
-    path: &CanonicalPath,
-    metadata: Metadata,
-) -> Result<IndexEntry, Error> {
+fn scan_entry(path: &CanonicalPath, metadata: Metadata) -> Result<IndexEntry> {
     if metadata.is_dir() {
-        return Err(Error::msg("DirEntry is directory"));
+        return Err(ArklibError::Path);
     }
 
     let size = metadata.len();
     if size == 0 {
-        return Err(Error::msg("Empty resource"));
+        return Err(ArklibError::Path);
     }
```

Comment:
Looks like a problem with the content, not with the path.

Reply:
I used std::io::Error::new(std::io::Errorkind::InvalidData